Products/ZenModel/browser/Autocompleters.py

```diff
@@ -34,12 +34,11 @@ def __call__(self, query='', dataRoot='devices'):
         if dataRoot != 'devices':
             import exceptions
             raise exceptions.ValueError("dataRoot should only be 'devices'")
-        catalog = getToolByName(self.context.dmd.Devices,
-            self.context.dmd.Devices.default_catalog)
-        query = MatchGlob('titleOrId', query.rstrip('*') + '*')
+        catalog = IModelCatalogTool(self.context.dmd).devices
```

Comment:
how about:

if isinstance(self.context, DeviceOrganizer):
    catalog = IModelCatalogTool(self.context).devices
else:
    catalog = IModelCatalogTool(self.context.dmd.Devices).devices

here, and then get rid of lines 39-40 below